endpoints/zones/src/main.cpp

```diff
@@ -9,78 +9,468 @@
+				logging::info("{}: client_info.username = [{}]", fn, client_info.username);
+
+				http::response<http::string_body> res{http::status::ok, _req.version()};
+				res.set(http::field::server, irods::http::version::server_name);
+				res.set(http::field::content_type, "application/json");
+				res.keep_alive(_req.keep_alive());
+
+				try {
+					const auto name_iter = _args.find("name");
+					if (name_iter == std::end(_args)) {
+						logging::error("{}: Missing [name] parameter.", fn);
+						return _sess_ptr->send(irods::http::fail(res, http::status::bad_request));
+					}
 
-			BytesBuf* bbuf{};
-			irods::at_scope_exit free_bbuf{[&bbuf] { freeBBuf(bbuf); }};
+					const auto property_iter = _args.find("property");
+					if (property_iter == std::end(_args)) {
+						logging::error("{}: Missing [property] parameter.", fn);
+						return _sess_ptr->send(irods::http::fail(res, http::status::bad_request));
+					}
 
-			{
-				auto conn = irods::get_connection(client_info.username);
+					const auto value_iter = _args.find("value");
+					if (value_iter == std::end(_args)) {
+						logging::error("{}: Missing [value] parameter.", fn);
+						return _sess_ptr->send(irods::http::fail(res, http::status::bad_request));
+					}
 
-				if (const auto ec = rcZoneReport(static_cast<RcComm*>(conn), &bbuf); ec != 0) {
-					logging::error("{}: rcZoneReport error: [{}]", fn, ec);
-					return _sess_ptr->send(irods::http::fail(status_type::bad_request));
+					auto conn = irods::get_connection(client_info.username);
+
+					if (property_iter->second == "name") {
+						adm::client::modify_zone(conn, name_iter->second, adm::zone_name_property{value_iter->second});
+					}
+					else if (property_iter->second == "connection_info") {
+						adm::client::modify_zone(
+							conn, name_iter->second, adm::connection_info_property{value_iter->second});
+					}
+					else if (property_iter->second == "comment") {
+						adm::client::modify_zone(conn, name_iter->second, adm::comment_property{value_iter->second});
+					}
+					else if (property_iter->second == "acl") {
+						// The string referenced by the value iterator is expected to have the following structure:
+					    //
+					    //     <acl>:<user>
+					    //
+						const std::string_view acl_user_pair = value_iter->second;
+
+						// Extract ACL from input string.
+						const auto colon_pos = acl_user_pair.find(':');
+						if (std::string::npos == colon_pos) {
+							logging::error(
+								"{}: Invalid value for [value] parameter. Received [{}]. Missing colon delimiter.",
+								fn,
+								acl_user_pair);
+							return _sess_ptr->send(irods::http::fail(res, http::status::bad_request));
+						}
+
+						const auto acl_sv = acl_user_pair.substr(0, colon_pos);
+
+						// Extract username from input string.
+					    // TODO Is there a safer way to check this? It's possible the string could exceed the maximum
+					    // allowed size.
+						if (colon_pos + 1 >= acl_user_pair.size()) {
```

Comment:
This isn't so bad given the fact that the HTTP API enforces a max size on the request body.

There's probably a better way to enforce this though.

Reply:
Removed.